samcli/lib/build/workflow_config.py

```diff
@@ -116,31 +121,28 @@ def get_selector(selector_list, identifiers, specified_workflow=None):
     """
 
     # Create a combined view of all the selectors
-    all_selectors = {}
+    all_selectors: Dict[str, Union["BasicWorkflowSelector", "ManifestWorkflowSelector"]] = dict()
     for selector in selector_list:
         all_selectors = {**all_selectors, **selector}
 
     # Check for specified workflow being supported at all and if it's not, raise an UnsupportedBuilderException.
     if specified_workflow and specified_workflow not in all_selectors:
         raise UnsupportedBuilderException("'{}' does not have a supported builder".format(specified_workflow))
 
-    # Loop through all identifers to gather list of selectors with potential matches.
-    selectors = [all_selectors.get(identifier, None) for identifier in identifiers]
-
-    # Intialize a `None` selector.
-    selector = None
+    # Loop through all identifiers to gather list of selectors with potential matches.
+    selectors = [all_selectors.get(identifier) for identifier in identifiers if identifier]
 
     try:
         # Find first non-None selector.
         # Return the first selector with a match.
-        selector = next(_selector for _selector in selectors if _selector)
+        return next(_selector for _selector in selectors if _selector)
     except StopIteration:
         pass
 
-    return selector
+    return None
```

Comment:
I prefer the previous version (return selector), is there a reason to change it?

Reply:
two reasons:
1. I don't need to annotate `selector = None` to `selector: Optional[.....] = None`
2. return inside try matches the comment "Return the first selector with a match."